core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2MetadataClient.java

```diff
@@ -266,26 +276,31 @@ public Ec2MetadataBuilder endpointMode(EndpointMode endpointMode) {
         @Override
         public Ec2MetadataBuilder httpClient(SdkHttpClient httpClient) {
             this.httpClient = httpClient;
+            this.httpClientIsInternal = false;
             return this;
         }
 
         @Override
+        public Builder httpClient(DefaultSdkHttpClientBuilder builder) {
+            SdkHttpClient builtHttpClient = builder.build();
+            this.httpClientIsInternal = true;
+            this.httpClient = builtHttpClient;
+            return this;
+        }
+
         public Ec2MetadataRetryPolicy getRetryPolicy() {
             return this.retryPolicy;
         }
 
-        @Override
-        public URI getEndpoint() {
+        URI getEndpoint() {
```

Comment:
We should probably make them public to be consistent with other getters used for reflection.